app/User.php

```diff
@@ -59,7 +60,21 @@ public function printJobs()
         return $this->hasMany('App\PrintJob');
     }
 
+    public function roles() {
+        return $this->belongsToMany(Role::class, 'role_users');
+    }
+
+    public function inRole(array $roleNames, $objectId = null)
+    {
+        return DB::table('role_users')
+                ->join('roles', 'roles.id', '=', 'role_users.role_id')
+                ->where('role_users.user_id', '=', $this->id)
+                ->whereIn('roles.name', $roleNames)
+                ->where('role_users.object_id', '=', $objectId)
+                ->count() > 0;
+    }
```

Comment:
hasRole or even better hasAnyRole instead of inRole?

I think we should not query the DB every time when checking roles, just once per request. Also, we should maybe use the relationship not raw join for this.
Can we find a better implementation of this core function?